portal/server/utils/csv-utils.js

```diff
@@ -41,6 +42,25 @@ const excelColumnToColumnIndex = (column) => {
   return result - 1;
 };
 
+/**
+ * Checks for floating-point rounding errors and rounds to two decimal places if within error tolerance.
+ * @param {number} number - The number to check and potentially round.
+ * @returns {number} - The rounded number to two decimal places if within tolerance, otherwise the original number.
+ * @throws {TypeError} - If the input is not a number.
+ */
+const handleFloatingPointRoundingErrors = (number) => {
```

Comment:
also could we add (number): number to give the type of it properly please

Reply:
JS file so can't add type annotations - the param and return types are specified in the function doc above though so the IDE should pick this up

Reply:
he he I'm blind, sorry